### DISCUSSION TITLE
Gradle plugin

### DISCUSSION THREAD

**Discussion by JavierSegoviaCordoba** [facebook/docusaurus#7797] [Ideas]
I have been talking with @cortinico about possible integration with Gradle.

Tasks which a Docusaurus Gradle plugin can provide:

- `docusaurusInit` - generate initial site
- `docusaurusServe` - live reload server
- `docusaurusBuild` - build documentation
- `docusaurusPublish` - publish to GitHub pages

**Comment by cortinico** [comment]
Thanks for opening this @JavierSegoviaCordoba 

> Tasks which a Docusaurus Gradle plugin can provide:

I thought a bit about this. I'm actually wondering if this can easily be achieved with: `gradle-node-plugin` (see https://github.com/node-gradle/gradle-node-plugin/blob/master/docs/usage.md) without the need of any specific plugin.

**Reply by slorber** [reply]
Hey

I'm not sure we'd want to maintain in Docusaurus core a plugin for every build system out there (Gradle, Maven, SBT...). That would be too much work on a lot of various technologies that we couldn't easily maintain over time with our small team.

It looks to me that all these build tools have something to run npm scripts, and generic tools are likely good enough.

----

I'm not sure see the value of running Docusaurus in the middle of an existing Gradle build script anyway.

Couldn't you just do this?

```bash
gradlew assembleRelease
npm run build
```

The only advantage of the integration would be to be able to interleave Docusaurus commands in the middle of something else run by Gradle, and I'd like to understand better when you'd really want to do that in the first place 🤷‍♂️ . Afaik it is possible to run bash scripts from Gradle so you can execute npm commands quite easily (or use the gradle node plugin).

Otherwise, if it's just for running Docusaurus commands at the beginning or the end, is the integration really worth it?

**Reply by JavierSegoviaCordoba** [reply]
I like to have everything configured via Gradle so if I change from one CI provider, the migration is usually simpler, anyway, I understand if resources are limited this can be out.
